account_asset_management/models/account_asset_line.py

```diff
@@ -0,0 +1,331 @@
+# Copyright 2009-2018 Noviat
+# Copyright 2021 Tecnativa - João Marques
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import UserError
+
+
+class AccountAssetLine(models.Model):
+    _name = "account.asset.line"
+    _description = "Asset depreciation table line"
+    _order = "type, line_date"
+    _check_company_auto = True
+
+    name = fields.Char(string="Depreciation Name", size=64, readonly=True)
+    asset_id = fields.Many2one(
+        comodel_name="account.asset",
+        string="Asset",
+        required=True,
+        ondelete="cascade",
+        check_company=True,
+        index=True,
+    )
+    previous_id = fields.Many2one(
+        comodel_name="account.asset.line",
+        string="Previous Depreciation Line",
+        readonly=True,
+    )
+    parent_state = fields.Selection(
+        related="asset_id.state", string="State of Asset", readonly=True
+    )
+    depreciation_base = fields.Float(
+        related="asset_id.depreciation_base", string="Depreciation Base", readonly=True
+    )
+    amount = fields.Float(digits="Account", required=True)
+    remaining_value = fields.Float(
+        compute="_compute_values",
+        digits="Account",
+        string="Next Period Depreciation",
+        store=True,
+    )
+    depreciated_value = fields.Float(
+        compute="_compute_values",
+        digits="Account",
+        string="Amount Already Depreciated",
+        store=True,
+    )
+    line_date = fields.Date(string="Date", required=True)
+    line_days = fields.Integer(string="Days", readonly=True)
+    move_id = fields.Many2one(
+        comodel_name="account.move",
+        string="Depreciation Entry",
+        readonly=True,
+        check_company=True,
+    )
+    move_check = fields.Boolean(
+        compute="_compute_move_check", string="Posted", store=True
+    )
+    type = fields.Selection(
+        selection=[
+            ("create", "Depreciation Base"),
+            ("depreciate", "Depreciation"),
+            ("remove", "Asset Removal"),
+        ],
+        readonly=True,
+        default="depreciate",
+    )
+    init_entry = fields.Boolean(
+        string="Initial Balance Entry",
+        help="Set this flag for entries of previous fiscal years "
+        "for which Odoo has not generated accounting entries.",
+    )
+    company_id = fields.Many2one(
+        "res.company",
```

Comment:
no need to re-define model for related field

Reply:
You mean not adding "res.company" in the field definition? however, this is also contemplated in pr #1513.

Reply:
hi i add pr https://github.com/OCA/account-financial-tools/pull/1513 and now it ok